.claude/hooks/post-tool-call.py

```
@@ -0,0 +1,93 @@
+import traceback
+from contextlib import closing
+from typing import Optional
+
+WEBSERVER_HOST = "localhost"
+WEBSERVER_ENDPOINT = "/api/provenance/call"
+PORT_FILE_SUFFIX = "-provenance-port.txt"
+
+class ProvenanceHookError(RuntimeError):
+    pass
+
+def http_request(method, host, port, location, *, body: Optional[bytes] = None, headers={}, timeout=None) -> bytes:
```

Ruff (0.14.14)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)